packages/jest-core/src/SearchSource.ts

```diff
@@ -263,7 +262,21 @@ export default class SearchSource {
         changedFiles,
         globalConfig.collectCoverage,
       );
-    } else if (globalConfig.runTestsByPath && paths && paths.length) {
+    }
+
+    let paths = globalConfig.nonFlagArgs;
+
+    if (globalConfig.findRelatedTests && 'win32' === os.platform()) {
```

Comment:
mac by default is also case insensitive, but I'm not sure if it's possible to detect it somehow beyond checking the os?

Reply:
Unfortunately I don't know any api that could be used to check it